core/src/main/java/org/infinispan/reactive/publisher/impl/PublisherHandler.java

```diff
@@ -0,0 +1,567 @@
+package org.infinispan.reactive.publisher.impl;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Function;
+
+import org.infinispan.commons.util.IntSet;
+import org.infinispan.commons.util.IntSets;
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.factories.KnownComponentNames;
+import org.infinispan.factories.annotations.ComponentName;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.factories.annotations.Start;
+import org.infinispan.factories.annotations.Stop;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+import org.infinispan.notifications.Listener;
+import org.infinispan.notifications.cachemanagerlistener.CacheManagerNotifier;
+import org.infinispan.notifications.cachemanagerlistener.annotation.ViewChanged;
+import org.infinispan.notifications.cachemanagerlistener.event.ViewChangedEvent;
+import org.infinispan.reactive.publisher.impl.commands.batch.InitialPublisherCommand;
+import org.infinispan.reactive.publisher.impl.commands.batch.KeyPublisherResponse;
+import org.infinispan.reactive.publisher.impl.commands.batch.PublisherResponse;
+import org.infinispan.remoting.transport.Address;
+import org.infinispan.util.concurrent.CompletableFutures;
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import io.reactivex.Flowable;
+import io.reactivex.FlowableSubscriber;
+import io.reactivex.functions.Consumer;
+import io.reactivex.internal.subscriptions.SubscriptionHelper;
+import net.jcip.annotations.GuardedBy;
+
+/**
+ * Handler for holding publisher results between requests of data
+ * @since 10.1
+ */
+@Scope(Scopes.NAMED_CACHE)
+@Listener(observation = Listener.Observation.POST)
+public class PublisherHandler {
+   private static final Log log = LogFactory.getLog(MethodHandles.lookup().lookupClass());
+   private static final boolean trace = log.isTraceEnabled();
+
+   private final ConcurrentMap<Object, PublisherState> currentRequests = new ConcurrentHashMap<>();
+
+   @Inject CacheManagerNotifier managerNotifier;
+   @Inject @ComponentName(KnownComponentNames.ASYNC_OPERATIONS_EXECUTOR)
+   ExecutorService cpuExecutor;
+   @Inject LocalPublisherManager lpm;
```

Comment:
Nitpick, but `localPublisherManager` is used consistently elsewhere.

Reply:
Refactored.